sdk/src/wrapped_azero.ts

```diff
@@ -0,0 +1,118 @@
+import { ApiPromise } from '@polkadot/api'
+import { ContractPromise } from '@polkadot/api-contract'
+import { WeightV2 } from '@polkadot/types/interfaces'
+import { IKeyringPair } from '@polkadot/types/types'
+import { DeployedContract } from '@scio-labs/use-inkathon'
+import { deployContract } from '@scio-labs/use-inkathon/helpers'
+import { Network } from './network.js'
+import { PSP22Query, PSP22Tx, WrappedAZEROTx } from './schema.js'
+import { DEFAULT_PROOF_SIZE, DEFAULT_REF_TIME, sendQuery, sendTx } from './utils.js'
+
+export class WrappedAZERO {
+  contract: ContractPromise | null = null
+  api: ApiPromise
+  gasLimit: WeightV2
+  storageDepositLimit: number | null
+  waitForFinalization: boolean
+
+  constructor(
+    api: ApiPromise,
+    network: Network,
+    storageDepositLimit: number | null = null,
+    refTime: number = DEFAULT_REF_TIME,
+    proofSize: number = DEFAULT_PROOF_SIZE
+  ) {
+    this.api = api
+    this.gasLimit = api.registry.createType('WeightV2', {
+      refTime,
+      proofSize
+    }) as WeightV2
+    this.storageDepositLimit = storageDepositLimit
+    this.waitForFinalization = network != Network.Local
+  }
+
+  async deploy(account: IKeyringPair, abi: any, wasm: Buffer): Promise<DeployedContract> {
+    return deployContract(this.api, account, abi, wasm, 'new', [])
+  }
+
+  async load(deploymentAddress: string, abi: any): Promise<void> {
+    this.contract = new ContractPromise(this.api, abi, deploymentAddress)
+  }
+
+  async deposit(account: IKeyringPair, value: number): Promise<string> {
+    return sendTx(
+      this.contract,
+      this.gasLimit,
+      this.storageDepositLimit,
+      value,
+      account,
+      WrappedAZEROTx.Deposit,
+      [],
+      this.waitForFinalization
+    )
+  }
+
+  async depositWithoutFinalization(account: IKeyringPair, value: number): Promise<string> {
+    return sendTx(
+      this.contract,
+      this.gasLimit,
+      this.storageDepositLimit,
+      value,
+      account,
+      WrappedAZEROTx.Deposit,
+      [],
+      this.waitForFinalization,
+      false
+    )
+  }
+
+  async withdraw(account: IKeyringPair, value: number): Promise<string> {
+    return sendTx(
+      this.contract,
+      this.gasLimit,
+      this.storageDepositLimit,
+      0,
+      account,
+      WrappedAZEROTx.Withdraw,
+      [value],
+      this.waitForFinalization
+    )
+  }
+
+  async withdrawWithoutFinalization(account: IKeyringPair, value: number): Promise<string> {
```

Comment:
We should drop approach with separate functions without finalization. Just expose optional `block` param in base function such as `withdraw`